src/codewhisperer/commands/onInlineAcceptance.ts

```diff
@@ -116,10 +116,12 @@ export async function onInlineAcceptance(
             completionType: acceptanceEntry.completionType,
             language: languageContext.language,
         })
-        const codeRangeAfterFormat = new vscode.Range(start, acceptanceEntry.editor.selection.active)
+        // use the effective range of accepted recommendation
+        // to avoid counting typeahead in accepted tokens
+        const insertedCoderange = new vscode.Range(acceptanceEntry.effectiveRange.start, end)
         CodeWhispererCodeCoverageTracker.getTracker(languageContext.language, globalStorage)?.countAcceptedTokens(
-            codeRangeAfterFormat,
-            acceptanceEntry.editor.document.getText(codeRangeAfterFormat),
+            insertedCoderange,
```

Comment:
I thought in JB we still count typeahead?

Reply:
Have we checked JB has the same implementation as this?

Reply:
I will make another PR in JetBrains to align this. 

Typeahead is user input code, not written by CWSPR. So this implementation here is correct. If JB is otherwise, I will fix it.